obspy/io/mseed/tests/test_mseed_special_issues.py

```diff
@@ -1163,6 +1166,61 @@ def test_reading_file_larger_than_2048_mib(self, getsize_mock):
             "read the file in chunks as documented here: "
             "https://github.com/obspy/obspy/pull/1419#issuecomment-221582369")
 
+    def test_read_file_with_non_valid_blocks_in_between(self):
+        """
+        Test reading MiniSEED files that have some non-valid blocks in-between.
+        """
+        # This file has two 4096 bytes records.
+        filename = os.path.join(self.path, 'data', 'test.mseed')
+        with io.open(filename, "rb") as fh:
+            rec1 = fh.read(4096)
+            rec2 = fh.read(4096)
+
+        reference = _read_mseed(filename)
+        del reference[0].stats.mseed
+
+        # Fill with zero bytes.
+        for length in (128, 256, 512, 1024, 2048, 4096, 8192):
+            with io.BytesIO() as buf:
+                buf.write(rec1)
+                buf.write(b'\x00' * length)
+                buf.write(rec2)
+                buf.seek(0, 0)
+                # This will raise 1 warning per 128 bytes.
+                with WarningsCapture() as w:
+                    st = _read_mseed(buf)
+                self.assertEqual(len(w), length // 128)
+
```

Comment:
Maybe confirm the expected number of records in the stream too in this test.

Reply:
Done